interview-questions.md

```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is an instance variable in Ruby? How is it different from other variables?
 
-  Your answer:
+  Your answer:  the instance variable (declared with @) in ruby is declared when you create a class. That will allow every object that is an instance of that class to have access to that variable. It becomes part of the objects.
 
-  Researched answer:
+  Researched answer: to add to my answer, the objects will have access to the intance variable declared in the class but each object (instance of the class) can have different values for theis instance variables.
 
 
 
 2. What is a block in Ruby?
 
-  Your answer:
+  Your answer: a block is a piece of code that is performing a task. Much like a function in JS.
 
-  Researched answer:
+  Researched answer: Blocks are like anonymous functions in JS. We create blocks by starting with 'do' and ending with 'end'. We can also use {} to create a block. Blocks are a set of instructions (code) that is executed by method. 
 
 
 
 3. Ruby has an implicit return. What does that mean?
 
-  Your answer:
+  Your answer: that we do not need the keyword 'return' to return the result from a block or statement. it is implicit that the function or block will be returning something.
 
-  Researched answer:
+  Researched answer: it means that the last line inside of a method will always be returned (without the need of the return keyword)
 
 
 
 4. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: OOP is a programing paradigm. It is a set of rules that was agreed upon programers to structure the code. The four pillars that support the OOP paradigm are encapsulation, abstraction, inheritance and polymorphism. In functional programming we have the logic wrapped in functions that can be called as needed. That can become very complex / difficult to read / make changes. In oop we define the logic inside of classes and objects and this allows us to easily find the pieace of code that we would like to change, change that code without loosing sleep at night, or add a functionality without having to repeat your code. With functional programing we have our functions and variables 'floating around'. 
```

Comment:
This is a great answer to hang on to if you ever get a chance to use it in an interview!